pkg/cmd/backuprestore/backupnoconfig.go

```diff
@@ -0,0 +1,246 @@
+	return cmd
+}
+
+func (b *backupNoConfig) AddFlags(fs *pflag.FlagSet) {
+	b.backupOptions.AddFlags(fs)
+}
+
+func (b *backupNoConfig) Validate() error {
+	return b.Validate()
+}
+
+func (b *backupNoConfig) Run() error {
+	backupsClient, err := b.getBackupClient()
+	if err != nil {
+		return err
+	}
+
+	if err = b.extractBackupSpecs(backupsClient); err != nil {
+		return err
+	}
+
+	var errs []error
+	go func() {
+		err := b.scheduleBackup()
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}()
+	go func() {
+		err := b.scheduleBackupPrune()
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}()
+
+	if len(errs) > 0 {
+		return kerrors.NewAggregate(errs)
+	}
+	return nil
+}
+
+func (b *backupNoConfig) getBackupClient() (backupv1client.BackupsGetter, error) {
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", b.kubeConfig)
+	if err != nil {
+		bErr := fmt.Errorf("error loading kubeconfig: %v", err)
+		klog.Error(bErr)
+		return nil, bErr
+	}
+
+	backupsClient, err := backupv1client.NewForConfig(kubeConfig)
+	if err != nil {
+		bErr := fmt.Errorf("error creating etcd backups client: %v", err)
+		klog.Error(bErr)
+		return nil, bErr
+	}
+
+	return backupsClient, nil
+}
+
+func (b *backupNoConfig) extractBackupSpecs(backupsClient backupv1client.BackupsGetter) error {
+	backups, err := backupsClient.Backups().List(context.Background(), v1.ListOptions{})
+	if err != nil {
+		lErr := fmt.Errorf("could not list backup CRDs, error was: [%v]", err)
+		klog.Error(lErr)
+		return lErr
+	}
+
+	if len(backups.Items) == 0 {
+		lErr := fmt.Errorf("no backup CRDs exist, found [%v]", backups)
+		klog.Error(lErr)
+		return lErr
+	}
+
+	idx := slices.IndexFunc(backups.Items, func(backup backupv1alpha1.Backup) bool {
+		return backup.Name == "default"
+	})
+	if idx == -1 {
+		sErr := fmt.Errorf("could not find default backup CR, found [%v]", backups.Items)
+		klog.Error(sErr)
+		return sErr
+	}
+
+	defaultBackupCR := backups.Items[idx]
+	b.schedule = defaultBackupCR.Spec.EtcdBackupSpec.Schedule
+	b.retention = defaultBackupCR.Spec.EtcdBackupSpec.RetentionPolicy
+
+	return nil
+}
+
+func (b *backupNoConfig) backup() error {
+	// initially take backup using etcdctl
+	if !b.snapshotExist {
+		if err := backup(&b.backupOptions); err != nil {
+			klog.Errorf("run: backup failed: [%v]", err)
+			return err
+		}
+		b.snapshotExist = true
+		klog.Infof("config-dir is: %s", b.configDir)
+		return nil
+	}
+
+	// only update the snapshot file
+	if err := b.copySnapshot(); err != nil {
+		sErr := fmt.Errorf("run: backup failed: [%v]", err)
+		klog.Error(sErr)
+		return sErr
+	}
+
+	return nil
+}
+
+func (b *backupNoConfig) scheduleBackup() error {
+	s, _ := gcron.NewScheduler()
+	defer func() { _ = s.Shutdown() }()
+
+	if _, err := s.NewJob(
+		gcron.CronJob(
+			b.schedule,
+			false,
+		),
+		gcron.NewTask(b.backup()),
+	); err != nil {
+		return err
+	}
+
+	s.Start()
+	return nil
+}
+
+func (b *backupNoConfig) copySnapshot() error {
```

Comment:
can we please remove this? I think this is going to confuse people more than this is actually helping

Reply:
done 👍🏽